joern-cli/frontends/gosrc2cpg/src/main/scala/io/joern/gosrc2cpg/astcreation/CacheBuilder.scala

```diff
@@ -120,19 +119,25 @@ trait CacheBuilder(implicit withSchemaValidation: ValidationMode) { this: AstCre
   }
 
   protected def processTypeSepc(typeSepc: Value): (String, String, Seq[Ast]) = {
-    val name     = typeSepc(ParserKeys.Name)(ParserKeys.Name).str
-    val fullName = fullyQualifiedPackage + Defines.dot + name
-    val typeNode = createParserNodeInfo(typeSepc(ParserKeys.Type))
-    val ast = typeNode.node match {
-      // As of don't see any use case where InterfaceType needs to be handled.
-      case InterfaceType => Seq.empty
-      // astForStructType() function will record the member types
-      case StructType => astForStructType(typeNode, fullName)
-      // Process lambda function types to record lambda function signature mapped to TypeFullName
-      case FuncType => processFuncType(typeNode, fullName)
-      case _        => Seq.empty
-    }
-    (name, fullName, ast)
+    val name = typeSepc(ParserKeys.Name)(ParserKeys.Name).str
+    if (!goGlobal.processingDependencies || goGlobal.processingDependencies && name.headOption.exists(_.isUpper)) {
```

Comment:
This check is duplicated 3 times, I wonder if you could potentially make a function of this explaining the check it's looking for in one place?

Reply:
I see it's duplicated more below, I think it should definitely be a function